### ISSUE TITLE
RKE2/k3s Provisioning Tests are stuck on k8s version 1.22.x; need to be running on the latest instead

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#40054]
RKE2/k3s provisioning tests are currently testing downstream clusters `v1.22.16+k3s1` despite default for 2.7.x being v1.24.x for quite a while (and recently bumped to v1.25.x).

This is because the script to determine the version to test on gets the `latest` version from the first channel:
https://github.com/rancher/rancher/blob/0ba317b31ece54ee115ec52b927a30d0fe6813c2/scripts/provisioning-tests#L42

This was acceptable a while ago, but since we implemented Rancher-version specific defaults per https://github.com/rancher/rancher/issues/36827 the actual defauls are in `appDefaults` section in KDM: https://github.com/rancher/kontainer-driver-metadata/blob/5805ac7615ac773f2dde432799556a0cc3117c9e/channels-rke2.yaml#L4-L14

And the logic to determine default k3s/RKE2 version can be found here:
https://github.com/rancher/rancher/blob/33bd193b82afd25a18639b5918ccdc93a9bb7e4b/pkg/channelserver/channelserver.go#L144

It's suggested to update the `provisioning-tests` script to determine the k8s version to test on based on settings in `appDefaults`. Note that the logic can be significantly simplified by grabbing the last element in `appDefaults` as it's what the "latest released / to be released" Rancher version will be falling under in the respective branch in KDM, e.g. here it is for `dev-v2.6`: https://github.com/rancher/kontainer-driver-metadata/blob/1bbe1c16885d75c2ae1e1edeb005bca0613d53d0/channels-rke2.yaml#L4-L12

**Comment by sbstp** [comment]
@snasovich I think this would be a good candidate for backport, do you agree?

**Comment by snasovich** [comment]
@sbstp , good catch. Logged https://github.com/rancher/rancher/issues/40109 for you to address, thank you.

**Comment by snasovich** [comment]
Confirming the latest available version is used in `release/v2.7` builds now. From https://drone-publish.rancher.io/rancher/rancher/8829/2/2:
```
Starting rancher server for provisioning-tests using v1.25.5+rke2r2
```
